File: WasteVisionWebBE/Domain/Users/UserDto.cs
using System.Text.Json.Serialization;
using DDDSample1.Domain.Roles;

namespace DDDSample1.Domain.Users
{
    public class UserDto
    {
        public string Id { get; set; }
        public string Email { get; set; }
        public string Username { get; set; }
        public string RoleId { get; set; }

        public UserDto()
        {
        }

        [JsonConstructor]
        public UserDto(string id, string email, string username, string roleId)
        {
            Id = id;
            Email = email;
            Username = username;
            RoleId = roleId;
        }

        public UserDto(string id, string email, string username, RoleId roleId)
        {
            Id = id;
            Email = email;
            Username = username;
            RoleId = roleId.Value;
        }
    }
}

File: WasteVisionWebFE/src/data/dto/user-dto.ts
import { Role } from "@/core/domain/Role";

export interface UserDTO {
  id?: string;
  email: string;
  username: string;
  roleId: string;
  roleName?: string;
  active: boolean;
  role?: Role;
}

export interface CreateUserDTO {
  email: string;
  username: string;
  roleId: string;
}

export interface UpdateUserDTO {
  email: string;
  username: string;
  roleId: string;
}

export interface UserSearchParamsDTO {
  email?: string;
  username?: string;
  roleId?: string;
}
